integrations/tensorflow/bindings/python/pyiree/tf/support/tf_utils.py

```diff
@@ -329,51 +339,62 @@ def __call__(self, *args, **kwargs):
-    iree_driver=None,
-    iree_compiler_targets=None)
-# tf_also is used for checking test consistency
-# to catch any initialization/randomization issues between model runs
-BackendInfo.add(
-    name="tf_also",
-    CompiledModule=TfCompiledModule,
-    iree_driver=None,
-    iree_compiler_targets=None)
-BackendInfo.add(
-    name="iree_vmla",
-    CompiledModule=IreeCompiledModule,
-    iree_driver="vmla",
-    iree_compiler_targets=["vmla"])
-BackendInfo.add(
-    name="iree_vulkan",
-    CompiledModule=IreeCompiledModule,
-    iree_driver="vulkan",
-    iree_compiler_targets=["vulkan-*"])
-BackendInfo.add(
-    name="iree_llvmjit",
-    CompiledModule=IreeCompiledModule,
-    iree_driver="llvm",
-    iree_compiler_targets=["llvm-ir"])
+    return tf.nest.map_structure(lambda t: self._convert_to_numpy(t)
+                                 if isinstance(t, tf.Tensor) else t,
+                                 *results,
+                                 check_types=False)
+
+
+class BackendInfo:
+
+  _NAME_TO_INFO = {
+      "tf": {
+          "compiled_module_class": TfCompiledModule,
+          "driver": None,
+          "compiler_targets": None,
+      },
+      "iree_vmla": {
+          "compiled_module_class": IreeCompiledModule,
+          "driver": "vmla",
+          "compiler_targets": ["vmla"]
+      },
+      "iree_llvmjit": {
+          "compiled_module_class": IreeCompiledModule,
+          "driver": "llvm",
+          "compiler_targets": ["llvm-ir"]
+      },
+      "iree_vulkan": {
+          "compiled_module_class": IreeCompiledModule,
+          "driver": "vulkan",
+          "compiler_targets": ["vulkan-*"]
+      },
+  }
+
+  def __init__(self, backend_name, artifact_name=None):
+    """Contains information for compiling the specified backend.
+
+    Args:
+      backend_name: a str specifying which backend to use. Should be one of
+        'tf', 'iree_vmla', 'iree_llvmjit', 'iree_vulkan'.
+      artifact_name: an optional str specifying what name to use when saving
+        compiled artifacts.
+    """
+    if backend_name not in self._NAME_TO_INFO:
+      raise KeyError(
+          "Expected backend_name to be one of "
+          f"{list(self._NAME_TO_INFO.keys())} but got '{backend_name}'.")
+    info = self._NAME_TO_INFO[backend_name]
+    self._compiled_module_class = info["compiled_module_class"]
+    self.driver = info["driver"]
+    self.compiler_targets = info["compiler_targets"]
+    self.name = backend_name if artifact_name is None else artifact_name
+
+  def compile(self, module, exported_names=(), artifacts_dir=None):
+    """Creates a `CompiledModule` for this backend."""
+    return self._compiled_module_class(module, self, exported_names,
+                                       artifacts_dir)
+
+  @staticmethod
+  def get_all_backends():
+    """Returns a list of all BackendInfo configurations."""
+    return [BackendInfo(backend_name) for backend_name in _BACKEND_NAME_TO_INFO]
```

Comment:
where is _BACKEND_NAME_TO_INFO defined?

Reply:
It wasn't – it was supposed to be renamed to `cls._name_to_info` after I moved that variable onto the class. Thanks for the catch!